discovery-provider/src/queries/queries.py

```diff
@@ -1982,3 +1983,121 @@ def get_top_genre_users():
+            Track.is_current == True,
+            Track.track_id == track_id
+        ).first()
+        
+        if (parent_track == None):
+            return api_helpers.error_response("Invalid track_id provided", 400)
+
+        track_owner_id = parent_track.owner_id
+        # Get the 'children' remix tracks
+        # Use the track owner id to fetch reposted/saved tracks returned first 
+        base_query = session.query(Track).join(
+            Remix,
+            and_(
+                Remix.child_track_id == Track.track_id,
+                Remix.parent_track_id == track_id
+            )
+        ).outerjoin(
+            Save,
+            and_(
+                Save.save_item_id == Track.track_id,
+                Save.save_type == SaveType.track,
+                Save.is_current == True,
+                Save.is_delete == False,
+                Save.user_id == track_owner_id
+            )
+        ).outerjoin(
+            Repost,
+            and_(
+                Repost.repost_item_id == Track.track_id,
+                Repost.user_id == track_owner_id,
+                Repost.repost_type == RepostType.track,
+                Repost.is_current == True,
+                Repost.is_delete == False
+            )
+        ).filter(
+            Track.is_current == True,
+            Track.is_delete == False
+        ).order_by(
+            desc(
+                case([
+                    (and_(Repost.created_at == None, Save.created_at == None), 0),
+                ],
+                else_ = 1)
+            ),
+            desc(
+                case([
+                    (or_(Repost.created_at == None, Repost.created_at <= Save.created_at), Save.created_at),
+                    (or_(Save.created_at == None, Repost.created_at >= Save.created_at), Repost.created_at)
+                ],
+                else_ = None)
+            ),
+            desc(Track.track_id)
+        )
+
+        tracks = paginate_query(base_query).all()
+        tracks = helpers.query_result_to_list(tracks)
+        track_ids = list(map(lambda track: track["track_id"], tracks))
+        current_user_id = get_current_user_id(required=False)
+        tracks = populate_track_metadata(session, track_ids, tracks, current_user_id)
+
+
+        if "with_users" in request.args and request.args.get("with_users") != 'false':
+            user_id_list = get_users_ids(tracks)
+            users = get_users_by_id(session, user_id_list)
+            for track in tracks:
+                user = users[track['owner_id']]
+                if user:
+                    track['user'] = user
+
+    return api_helpers.success_response(tracks)
+
+# Get the tracks that are 'parent' remixes of the requested track
+@bp.route("/remixes/<int:track_id>/parents", methods=("GET",))
+def get_remix_track_parents(track_id):
+    db = get_db_read_replica()
+    with db.scoped_session() as session:
+        base_query = session.query(
```

Comment:
Is it necessary to hit the `Remixes` table here? I thought we had all the info about parent remixes in the track metadata itself?

Reply:
Discussed offline, not necessary, but cleaner.